src/sentry/utils/metrics.py

```diff
@@ -271,12 +271,15 @@ def timer(
 
 
 def wraps(
-    key: str, instance: Optional[str] = None, tags: Optional[Tags] = None
+    key: str,
+    instance: Optional[str] = None,
+    tags: Optional[Tags] = None,
+    sample_rate: float = settings.SENTRY_METRICS_SAMPLE_RATE,
```

Comment:
Does using `settings` in a function definition work ok? Wasn't sure if it'd have problems since Django has to dynamically load settings, but maybe that happens before imports

if it doesn't like it you could just default to none and then override it when none in the function

Reply:
well we do it on the `timer` above. so happy to fix both or just stick with what we already have

Reply:
Ahh didn't see that, must be fine then. Good to know!